### ISSUE TITLE
How to replace text within a pre-existing PDF

### ISSUE THREAD

**Issue by alcukcoder** [jorisschellekens/borb#43]
Hi, this library looks really interesting. I'd like to learn it.
I couldn't see any documentation on how to replace existing text within an existing PDF, eg:

{{TITLE}} to Mr...

How would this be done?
Thank you :)

**Comment by jorisschellekens** [comment]
Replacing text is not trivial (regardless of the library).

You will encounter a problem called "reflowing the text". For instance, imagine if you had a piece of text where you want to replace a relatively short word "name" by a long one (for instance my name "Schellekens").

The text that follows may need to be placed differently. You have laid out ths text in columns, and suddenly the content needs to move to a different line.

That's very hard to do, considering this kind of structure information (which text belongs together, makes up a column of text, which lines make up a single paragraph, which lines belong to the same cell in a table, etc) is often missing in a PDF.

In short, replacing text is very very hard to do in PDF. You can only hope for the most basic of use-cases to work, like replacing text by a similarly long/short text. Or being allowed to change the font (so the text can at least be given the same width).

In borb, you can look for the bounding box of a piece of text. There's a tutorial on that in the examples repo.

You could then manually lay-out its replacement (on top of it, with a non-transparent background).

**Comment by alcukcoder** [comment]
OK this may be too difficult for my short project. Thanks anyway.
